samples/dialogs/dialogs.cpp

```diff
@@ -3117,6 +3089,27 @@ TestDefaultActionDialog::TestDefaultActionDialog( wxWindow *parent ) :
     grid_sizer->Add(new wxStaticText(this, wxID_ANY, "wxTextCtrl without wxTE_PROCESS_ENTER"),
                     wxSizerFlags().CentreVertical());
 
+    // We have to define a new class in order to actually handle pressing
+    // Enter, if we didn't do it, pressing it would still close the dialog.
+    class EnterHandlingTextCtrl : public wxTextCtrl
+    {
+    public:
+        EnterHandlingTextCtrl(wxWindow* parent, int id, const wxString& value)
+            : wxTextCtrl(parent, id, value,
+                         wxDefaultPosition, wxDefaultSize, wxTE_PROCESS_ENTER)
+        {
+            Bind(wxEVT_TEXT_ENTER, &EnterHandlingTextCtrl::OnEnter, this);
+
+            SetInitialSize(GetSizeFromTextSize(GetTextExtent(value).x));
+        }
+
+    private:
+        void OnEnter(wxCommandEvent& WXUNUSED(event))
```

Comment:
I wonder if it was not better in new code to use lambdas for one-line event handlers like this...

Reply:
Good point, thanks! Done now.